common/thorhelper/thorsoapcall.cpp

```diff
@@ -1897,6 +1897,15 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                             while(chunkSize && ch != '\n')//consume chunk-extension and CRLF
                                 dataProvider->getBytes(&ch, 1);
                         }
+                        //to support persistent connections we need to consume all trailing bytes for each message
+                        while ( ch != '\r' ) //consume trailers
```

Comment:
Is this the correct behaviour with the example:

```
7\r\n
chunks.\r\n  (data)
0\r\n        (final byte - 0)
\r\n         (end message)
```
Lines up to 1870 will read up to the 0, but line 1870/1871 will not skip the \r.  This code will then match and consume the \r\n after  the 0 but the end of message \r\n will not be consumed.

Reply:
I suspect I may have to consume one more crlf.  I'll take a look.

Reply:
An alternative is to remove the chunkSize test from line 1870.  I may be completely wrong - I only did a quick web search since I knew nothing about chunked encoding.

Reply:
I think the change will fix the common case.  It does not cope with the case where there is no output at all - so the total message is
```
0\r\n
\r\n
```
The spec also refers to needing to be able to skip trailing header fields

See https://datatracker.ietf.org/doc/html/rfc7230#section-4.1

The final termination looks like it is a single \r\n terminated line

Reply:
@ghalliday This loop is meant to consume those trailing headers (I was calling them "trailers"):
```
while ( ch != '\r' ) //consume trailers
{
  while (ch != '\n')
     dataProvider->getBytes(&ch, 1);
  dataProvider->getBytes(&ch, 1);
}
```
Basically if there are lines with content before the final close, consume those first.  Do you see any problems with it?  Wasn't able to really test.. and doubt it's used much.

Reply:
Yes you're right.  Maybe a comment before the while (indicating what it is skippoing), and another one before the next while
I think the chunkSize test in 1870 may also need to go to avoid problems with a completely zero length result.

Reply:
Ok I'll update.  But I'll let Vlad test as is as that won't test the zero length scenario anyway.

Reply:
Updated with fix for completely zero length result and added comments about consuming the trailing HTTP headers.

Test build should be deployed tomorrow morning and Vlad will test.